src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -1105,6 +1124,45 @@ static string mapPath(SourceReferenceResolver? referenceResolver, SyntaxTree tre
             {
                 return referenceResolver?.NormalizePath(tree.FilePath, baseFilePath: null) ?? tree.FilePath;
             }
+
+            // Caller must free the returned builder.
+            ArrayBuilder<string> getNamespaceNames()
```

Comment:
Is there an advantage to having allocation and free in different scopes? Consider allocating the `ArrayBuilder<string>` in the caller, and removing the comment.

Reply:
I'm satisfied with the current factoring though I know it adds a bit of mental overhead. Thanks